azext_iot/iothub/commands_message_endpoint.py

```diff
@@ -0,0 +1,227 @@
+    endpoint_policy_name: Optional[str] = None,
+    connection_string: Optional[str] = None,
+    authentication_type: Optional[str] = None,
+    endpoint_uri: Optional[str] = None,
+    entity_path: Optional[str] = None,
+    identity: Optional[str] = None,
+    resource_group_name: Optional[str] = None,
+):
+    message_endpoint_provider = MessageEndpoint(
+        cmd=cmd, hub_name=hub_name, rg=resource_group_name
+    )
+    return message_endpoint_provider.create(
+        endpoint_name=endpoint_name,
+        endpoint_type=EndpointType.ServiceBusQueue.value,
+        endpoint_account_name=endpoint_account_name,
+        endpoint_resource_group=endpoint_resource_group,
+        endpoint_subscription_id=endpoint_subscription_id,
+        endpoint_policy_name=endpoint_policy_name,
+        connection_string=connection_string,
+        authentication_type=authentication_type,
+        endpoint_uri=endpoint_uri,
+        entity_path=entity_path,
+        identity=identity
+    )
+
+
+def message_endpoint_create_service_bus_topic(
+    cmd,
+    hub_name: str,
+    endpoint_name: str,
+    endpoint_account_name: Optional[str] = None,
+    endpoint_resource_group: Optional[str] = None,
+    endpoint_subscription_id: Optional[str] = None,
+    endpoint_policy_name: Optional[str] = None,
+    connection_string: Optional[str] = None,
+    authentication_type: Optional[str] = None,
+    endpoint_uri: Optional[str] = None,
+    entity_path: Optional[str] = None,
+    identity: Optional[str] = None,
+    resource_group_name: Optional[str] = None,
+):
+    message_endpoint_provider = MessageEndpoint(
+        cmd=cmd, hub_name=hub_name, rg=resource_group_name
+    )
+    return message_endpoint_provider.create(
+        endpoint_name=endpoint_name,
+        endpoint_type=EndpointType.ServiceBusTopic.value,
+        endpoint_account_name=endpoint_account_name,
+        endpoint_resource_group=endpoint_resource_group,
+        endpoint_subscription_id=endpoint_subscription_id,
+        endpoint_policy_name=endpoint_policy_name,
+        connection_string=connection_string,
+        authentication_type=authentication_type,
+        endpoint_uri=endpoint_uri,
+        entity_path=entity_path,
+        identity=identity
+    )
+
+
+def message_endpoint_create_cosmos_db_collection(
+    cmd,
+    hub_name: str,
+    endpoint_name: str,
+    collection_name: str,
+    database_name: str,
+    endpoint_account_name: Optional[str] = None,
+    endpoint_resource_group: Optional[str] = None,
+    endpoint_subscription_id: Optional[str] = None,
+    connection_string: Optional[str] = None,
+    authentication_type: Optional[str] = None,
+    endpoint_uri: Optional[str] = None,
+    primary_key: Optional[str] = None,
+    secondary_key: Optional[str] = None,
+    partition_key_name: Optional[str] = None,
+    partition_key_template: Optional[str] = None,
+    identity: Optional[str] = None,
+    resource_group_name: Optional[str] = None,
+):
```

Comment:
@digimaun  @c-ryan-k endpoint_accountname should be renamed to cosmosdb_accountname and should not be optional. 

in reading the iot hub swagger:
      "required": [
        "name",  <----- account name
        "endpointUri",  <---- also are we concatenating the strings together to make the endpoint uri or should we make it also required?
        "databaseName",
        "collectionName"
      ]

Reply:
In the change set, the reason endpointUri is not required is because it can be explicitly provided, or derived from a connection string.

For "name" in the swagger, I believe that is referring to the endpoint name rather than the account name https://github.com/Azure/azure-rest-api-specs/blob/main/specification/iothub/resource-manager/Microsoft.Devices/preview/2022-04-30-preview/iothub.json#L3218

In the command the optional `--endpoint-account` is used to obtain a connection string for key based integration if `--connection-string` has not been provided. 

We should make sure help / errors are clear about the different flows.

Reply:
fixed error handling, the error now gives:
![image](https://user-images.githubusercontent.com/73560279/207697079-4b9f13fc-7e3b-45e7-95d1-fbff0724e0b3.png)

Reply:
@vilit1 just curiouis - if they provide both `--endpoint-account` and `-c` - do they get a `MutuallyExclusiveArgumentError` or do we just use the CS and swallow the endpoint account parameter?

Reply:
we use the account name from endpoint account and the keys from the connection string